sigstore/_internal/sct.py

```diff
@@ -16,75 +16,130 @@
 Utilities for verifying signed certificate timestamps.
 """
 
+import hashlib
+import logging
 import struct
+from typing import List, Union
 
-import cryptography.hazmat.primitives.asymmetric.ec as ec
+import cryptography.hazmat.primitives.asymmetric.padding as padding
 from cryptography.exceptions import InvalidSignature
 from cryptography.hazmat.primitives import hashes, serialization
-from cryptography.x509 import Certificate
+from cryptography.hazmat.primitives.asymmetric import ec, rsa
+from cryptography.x509 import Certificate, ExtendedKeyUsage, ObjectIdentifier
+from cryptography.x509.certificate_transparency import (
+    SignedCertificateTimestamp,
+)
 
-from sigstore._internal.fulcio import DetachedFulcioSCT
+logger = logging.getLogger(__name__)
 
 
-def _pack_digitally_signed(sct: DetachedFulcioSCT, cert: Certificate) -> bytes:
+def _cryptography_hash_to_enum(hash_: hashes.HashAlgorithm) -> int:
+    if type(hash_) is hashes.SHA256:
+        return 4
+    raise InvalidSctError(f"unexpected hash algorithm: {hash_!r}")
```

Comment:
Unused?

Reply:
Yep, that was from some earlier tests. I'll remove now.

Reply:
872bf8a..f8edf5e